mlir/tools/rocmlir-tuning-driver/rocmlir-tuning-driver.cpp

```diff
@@ -102,18 +102,19 @@ static benchmark::DataType getDataType(Type inputType) {
   }
 }
 
-// In order to match rocprof, returns time in nanoseconds
-static FailureOr<double> benchmarkKernel(const char *binary,
-                                         const char *funcName,
-                                         uint32_t blockSize, uint32_t gridSize,
-                                         benchmark::DataType dataType,
-                                         ArrayRef<size_t> bufferSizes) {
-  constexpr double msToNs = 1e6;
 // intentionally leaky macro
 #define HIPCHECK(expr)                                                         \
   if (hipSuccess != (expr)) {                                                  \
     return failure();                                                          \
   }
+
+// In order to match rocprof, returns time in nanoseconds
+static FailureOr<double>
+benchmarkKernel(const char *binary, const char *funcName, uint32_t blockSize,
+                uint32_t gridSize, benchmark::DataType dataType,
+                std::vector<void *> hostBuffers, std::vector<void *> gpuBuffers,
```

Comment:
Yeah, I had originally like this, but I "think" that `ArrayRef` is a const reference, but I have:
```
// HIP wants an array of pointers to each argument
  std::vector<void *> argPointers;
  for (void *&item : gpuBuffers) {
    argPointers.push_back(reinterpret_cast<void *>(&item));
  }
``` 
And this was making the compiler unhappy (because they should be `void const*` but I am casting to `void *`. `HostBuffer` can be an `ArrayRef`, but I thought nicer to have the same type for both (at the end it's a vector with three pointers and we afford copying it around). What do you think?

Reply:
Or (since someone in a different review reminded me of it), we could go with `std::array<void*, 3>&`

Reply:
I think the problem is : (void *&item : gpuBuffers). Here you iterate by reference rather than value -- so it can be problematic when the container is a const reference. I think if you iterate by value it shouldn't be an issue but then the saving is copying of the container vs copying of each item -- thus maybe not worth reworking in this case.